fflib/v1/buffer.go

```diff
@@ -66,13 +68,75 @@ type DecodingBuffer interface {
 	lener
 }
 
+var pools []*sync.Pool
+var pool64 *sync.Pool
+
+func init() {
+	var i int = 64
+	pools = make([]*sync.Pool, 14)
+	// TODO(pquerna): add science here.
+	for i = 6; i < 20; i++ {
+		n := int(math.Pow(2, float64(i)))
+		pools[poolNum(n)] = &sync.Pool{New: func() interface{} { return make([]byte, 0, n) }}
+	}
+	pool64 = pools[0]
+}
+
+func poolNum(i int) int {
+	// TODO(pquerna): convert to log2 w/ bsr asm instruction:
+	// 	<https://groups.google.com/forum/#!topic/golang-nuts/uAb5J1_y7ns>
+	if i <= 64 {
```

Comment:
Here is a function I converted to find the higest set bit value:

http://play.golang.org/p/31i0BvKczc